bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/VehicleHandler.java

```diff
@@ -0,0 +1,576 @@
+
+import javax.measure.quantity.Length;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.WWWAuthenticationProtocolHandler;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.MultiMap;
+import org.eclipse.jetty.util.UrlEncoded;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.openhab.binding.mercedesme.internal.utils.ChannelStateMap;
+import org.openhab.binding.mercedesme.internal.utils.Mapper;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.RawType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.storage.Storage;
+import org.openhab.core.storage.StorageService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.CommandOption;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VehicleHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class VehicleHandler extends BaseThingHandler {
+    private static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+    private static final String EXT_IMG_RES = "ExtImageResources_";
+    private static final String INITIALIZE_COMMAND = "Initialze";
+
+    private final Logger logger = LoggerFactory.getLogger(VehicleHandler.class);
+    private final Map<String, Long> timeHash = new HashMap<String, Long>();
+    private final MercedesMeCommandOptionProvider mmcop;
+    private final MercedesMeStateOptionProvider mmsop;
+    private final StorageService storageService;
+    private final HttpClient httpClient;
+    private final String uid;
+
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private Optional<AccountHandler> accountHandler = Optional.empty();
+    private Optional<QuantityType<?>> rangeElectric = Optional.empty();
+    private Optional<Storage<String>> imageStorage = Optional.empty();
+    private Optional<VehicleConfiguration> config = Optional.empty();
+    private Optional<QuantityType<?>> rangeFuel = Optional.empty();
+    private LocalDateTime nextRefresh;
```

Comment:
This should be `Instant`. Otherwise it will be susceptible to changes in system clock.

Reply:
Changed